torchaudio/functional/functional.py

```diff
@@ -1678,6 +1678,92 @@ def pitch_shift(
     return waveform_shift
 
 
+def _pitch_shift_preprocess(
```

Comment:
to avoid code duplication, you should use these helper functions in `pitch_shift` as well. this way, if we make a change to the implementation in the future, we do not need to remember to do it in two places

Reply:
also we generally unit test correctness for the functional version and do not have corresponding tests for transforms when the transform simply calls on the functional. Here you change the transforms to use these helper functions and but not for functional, so we don't actually know that the behavior of transforms wasn't changed. we can be more confident on that if functional and transforms have the same implementation, except one computes the kernel and one uses a cached one.